File: lab_2/IS_Lab2_CSharp/IS_Lab2_CSharp/JsonToYamlConverter.cs
using Newtonsoft.Json;
using YamlDotNet.Serialization;
using YamlDotNet.Serialization.NamingConventions;

namespace IS_Lab2_Extra;

public static class JsonToYamlConverter
{
    public static void Convert(List<Dictionary<string, object?>> deserializedData, string outputFilePath)
    {
        Console.WriteLine("Json to Yaml conversion (deserialized_data, destination_path)");
        var extractedData = DataExtractor.Extract(deserializedData);
        var serializer = new SerializerBuilder()
            .WithNamingConvention(CamelCaseNamingConvention.Instance)
            .Build();
        var yaml = serializer.Serialize(extractedData);
        var path = Path.Combine(AppContext.BaseDirectory, @"..\..\..\", outputFilePath);
        File.WriteAllText(path, yaml);
    }

    public static void Convert(string inputFilePath, string outputFilePath)
    {
        Console.WriteLine("Json to Yaml conversion (source_path, destination_path)");
        List<Dictionary<string, object?>> deserializedData;
        using (var r = new StreamReader(inputFilePath))
        {
            var json = r.ReadToEnd();
            deserializedData = JsonConvert.DeserializeObject<List<Dictionary<string, object?>>>(json)!;
        }

        var extractedData = DataExtractor.Extract(deserializedData);

        var serializer = new SerializerBuilder()
            .WithNamingConvention(CamelCaseNamingConvention.Instance)
            .Build();
        var yaml = serializer.Serialize(extractedData);
        var path = Path.Combine(AppContext.BaseDirectory, @"..\..\..\", outputFilePath);
        File.WriteAllText(path, yaml);
    }
}

File: lab_2/IS_Lab2_Python/json_to_yaml_converter.py
import json

import yaml
from multipledispatch import dispatch

from data_extractor import DataExtractor


class JsonToYamlConverter:

    @staticmethod
    @dispatch(str, str)
    def run(source_json_file_location, destination_yaml_file_location):
        print("Json to Yaml conversion (source_path, destination_path)")
        file = open(source_json_file_location, encoding="utf8")
        deserialized_data = json.load(file)
        extracted_data = DataExtractor.extract_from_json(deserialized_data)
        with open(destination_yaml_file_location, 'w', encoding='utf8') as f:
            yaml.dump(extracted_data, f, allow_unicode=True)

    @staticmethod
    @dispatch(list, str)
    def run(deserialized_data, destination_yaml_file_location):
        print("Json to Yaml conversion (deserialized_data, destination_path)")
        extracted_data = DataExtractor.extract_from_json(deserialized_data)
        with open(destination_yaml_file_location, 'w', encoding='utf8') as f:
            yaml.dump(extracted_data, f, allow_unicode=True)
